test/executor.test.js

```diff
@@ -906,6 +906,43 @@ describe('executor', function() {
     });
   });
 
+  describe.only('when booting with lazy connect', function() {
+    var DATASOURCE_THROWING_CONNECTION_ERROR = {
+      lazyConnector: {
+        connector: 'testLazyConnect',
+        name: 'lazyConnector',
+      },
+    };
+    var connectTriggered = true;
+
+    beforeEach(function() {
+      app.connector('testLazyConnect', {
+        initialize: function(dataSource, callback) {
+          if (dataSource.settings.lazyConnect) {
+            connectTriggered = false;
+          }
+        },
+      });
+    });
+
+    it('should trigger connect', function(done) {
+      boot.execute(app, someInstructions({
+        dataSources: DATASOURCE_THROWING_CONNECTION_ERROR,
+      }));
+      expect(connectTriggered).to.be.true;
+      done();
```

Comment:
Is it a good idea to rely on the fact that loopback-boot initializes datasources in the first tick? I think it will be safer to wait until loopback-boot is done:

```js
boot.execute(..., function(err) {
  if (err) return done(err);
  expect(connectTriggered)...
  done();
});
```

Thoughts?

Alternatively, if you prefer to stick with the sync way, then remove `done` from the tests as it's not needed:

```js
it('...', function() {
  boot.execute(...);
  expect(connectTriggered)...;
});